docs/reference/esql/functions/kibana/inline_cast.json

```diff
@@ -0,0 +1,19 @@
+{
+  "bool" : "boolean",
+  "boolean" : "boolean",
+  "cartesian_point" : "cartesian_point",
+  "cartesian_shape" : "cartesian_shape",
```

Comment:
I've noticed that these and the `geo` versions require that the value be a string. I.e. `123::cartesian_point` isn't allowed.

Are there any other constraints I should know about?

Reply:
It's actually short hand for a registered function. How about, instead of the result type, I add the name of the conversion function? That way you can use the list of valid function inputs in the other files.

Reply:
> It's actually short hand for a registered function.

So `123::cartesian_point` is syntactic sugar for `to_cartesian_point(123)`. 

> How about, instead of the result type, I add the name of the conversion function?

This sounds great.

Reply:
> So 123::cartesian_point is syntactic sugar for to_cartesian_point(123).

It's `to_cartesianpoint` in that case. It's not something you can do with text manipulation. It's two map lookups in a row. I'm not entirely sure why, but I'm sure there's a good reason. Either way, I can give you the mappings.

> This sounds great.

Done.